### DISCUSSION TITLE
Packaging for Linux

### DISCUSSION THREAD

**Discussion by VarLad** [GitoxideLabs/gitoxide#277] [Q&A] [answered]
Hey! To package this, do I need to distribute anything other than the `ein` and the `gix` binaries?

**Comment by Byron** [comment, marked as answer]
That's correct, this is all there is right now.

**Reply by VarLad** [reply]
Do I need to package all the `git-` crates well?

**Reply by Byron** [reply]
I don't know how the packaging is supposed to work in the distribution. Usually Rust uses `cargo` to handle its packages, so I'd assume that a binary distribution of the two discussed binaries would be enough.
For a source distribution, I'd hope it's possible to just use `cargo` to build on the target machine.

**Reply by VarLad** [reply]
@Byron The way it works here is, it builds the package with cargo first and the following make it into the package the user installs:
1. ein binary
2. gix binary
3. README.md
4. LICENSE

**Reply by VarLad** [reply]
So, I don't think the end users will get `git-` crates which are needed, if they install using the package?

**Reply by Byron** [reply]
The `git-*` crates are needed to build the binaries, but I'd hope it's something you don't have to package individually.

**Reply by VarLad** [reply]
@Byron Still somewhat confused, how do I version this? 
and from which of the source releases should I build?
Should I maintain a different build for git-odb and one for git-chunk and so on?

**Reply by Byron** [reply]
> @Byron Still somewhat confused, how do I version this?
> and from which of the source releases should I build?

I believe to be confused as well, maybe this can be untangled.

The version that matters for the binaries is the on of the `gitoxide` crate.

> Should I maintain a different build for git-odb and one for git-chunk and so on?

I'd hope this can be avoided. From my point of view all that's needed for binary releases is `cargo install gitoxide`. I do hope that the involved crates don't have to become part of the distribution's package manager as it sounds like a lot of work.

**Reply by VarLad** [reply]
@Byron I believe I was looking for https://github.com/Byron/gitoxide/releases/tag/v0.10.0 
Either way, the two binaries it produces is, `gixp` and `gix` (no sign of `ein`)
Is that good to go?

**Reply by Byron** [reply]
Ah, now I see. Which tags to use for binary releases, and it's the ones that don't have a `git-*` prefix.

`ein` seems new enough to not have been published yet, and I recommend avoiding publishing `0.10` knowing that the binary names will change.

There should be nothing preventing me to publish the latest `gitoxide` crate to help you get the latest and correctly named binaries.

**Reply by VarLad** [reply]
Thanks for that, I'll wait until the new version then!

**Reply by VarLad** [reply]
@Byron Looking at the Roadmap, any ETA for the (basic) features in 1.0? :)

**Reply by Byron** [reply]
I will be pushing hard to make it happen this year :).
